File: 21/21.py
#!/usr/bin/env python
from math import sqrt

def d(x):
  return sum(i + (x / i) for i in xrange(2, int(sqrt(x)) + 1) if x % i == 0) + 1

if __name__ == "__main__":
  answer = 0
  for a in xrange(1,10000):
    b = d(a)
    answer += a if d(b) == a and a != b else 0
  print "Problem 21 Answer:", answer

File: 21/21.c
#include <stdio.h>
#include <math.h>

int d(int x)
{
  int sumdiv = 1;
  int root = (int)sqrt(x) + 1;
  for (int i = 2; i <= root; i++) {
    if (x % i == 0) {
      sumdiv += (i + (x / i));
    }
  }
  return sumdiv;
}

int main(void)
{
  int answer = 0;
  int j;
  for (int i = 1; i < 10000; i++) {
    j = d(i);
    if (d(j) == i && j != i) {
      answer += i;
    }
  }
  printf("Problem 21 Answer: %d\n", answer);
}
